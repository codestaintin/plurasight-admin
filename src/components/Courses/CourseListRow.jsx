import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const CourseListRow = ({ course, handleDelete }) => (
  <tr>
    <td>
      <a
        href={course.watchHref}
        target="_blank"
        rel="noreferrer noopener"
      >
      Watch
      </a>
    </td>
    <td><Link to={`/course/${course.id}`}>{course.title}</Link></td>
    <td>{course.authorId}</td>
    <td>{course.category}</td>
    <td>{course.length}</td>
    <td>
      <button
        type="button"
        className="btn btn-danger"
        onClick={() => handleDelete(course.id)}
      >
        Delete
      </button>
    </td>
  </tr>
);

CourseListRow.propTypes = {
  course: PropTypes.object.isRequired,
  handleDelete: PropTypes.func
};

export default CourseListRow;
