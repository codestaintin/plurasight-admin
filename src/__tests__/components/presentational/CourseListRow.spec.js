import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import CourseListRow from '../../../components/Courses/CourseListRow';

configure({ adapter: new Adapter() });

describe('<CourseListRow>', () => {
  const props = {
    course: {
      id: '1'
    },
    handleDelete: jest.fn(() => {}),
  };

  it('renders the <CourseListRow> component', () => {
    const tree = shallow(<CourseListRow {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should simulate handleDelete', () => {
    const tree = shallow(<CourseListRow {...props} />);
    tree.find('button').simulate('click');
  });
});
