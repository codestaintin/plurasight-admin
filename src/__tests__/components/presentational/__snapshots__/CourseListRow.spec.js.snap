// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseListRow> renders the <CourseListRow> component 1`] = `
<tr>
  <td>
    <a
      rel="noreferrer noopener"
      target="_blank"
    >
      Watch
    </a>
  </td>
  <td>
    <Link
      replace={false}
      to="/course/1"
    />
  </td>
  <td />
  <td />
  <td />
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </td>
</tr>
`;
